src/common.h

```diff
@@ -0,0 +1,36 @@
+#ifndef COMMON_H
+#define COMMON_H
+
+#include "string.h"
+
+/*
+ * n represents size of string s. size of find is determinate by strlen
+ */
+char *strnstr(char *s, size_t n, char *find)
```

Comment:
can you put this function in correlated `.c` file? only declarations should be in header files I guess.